### DISCUSSION TITLE
Documentation for setting up a database outside of the container?

### DISCUSSION THREAD

**Discussion by itinneed2022** [graphite-project/graphite-web#2750] [Q&A]
Is there any documentation on how to connect a container-based graphite instance to a database running outside the container? Neither the quickstart or the rest of the documents seemed concerned with data loss in the event the container fails or is restarted.

**Comment by deniszh** [comment]
Hi @itinneed2022,

Usually containers do not loose data after restart, btw. If they did - the you will loose metric files also and setting up database outside of container will not help. Also database only store dashboards and events, if you not using them you can easily recreate it from scratch.
But if you still needed you can do that, of course it's possible. IMO setting up database outside of container is outside of scope of Graphite, you can use some 3rd party doc for that. Graphite-web using general Django database framework, you can use [Django doc](https://docs.djangoproject.com/en/2.0/ref/databases/) and connect it to some supported database (MySQL or Postgresql).
